### ISSUE TITLE
Provide small complete code samples

### ISSUE THREAD

**Issue by saun4app** [uikit/uikit#2804]
### Background

I have looked at [uikit](https://getuikit.com) many times, and am impressed by the beauty of `uikit` (technically and aesthetically).   `uikit` has all the good quality of a `css framework` that I look for:
- clean, concise syntax
- uses prefix like `uk-tab` instead of generic term, such as `tab`, allows a `css framework` to *co-exists* with other frameworks with little of no conflict. 
- offer extensive features
- aesthetically pleasing

I also agree with all the [PROs](https://www.slant.co/topics/150/viewpoints/20/~best-css-framework~uikit) of `uikit`.

An yet, I have not seriously play with `uikit` because I did not find many (any) code samples that are *not complete enough* to  get me started with seeing 'uikit' in action.  

Perhaps the lack of complete enough code samples pay heklp explain the [Not very popular CON](https://www.slant.co/topics/150/viewpoints/20/~best-css-framework~uikit) of `uikit`.

### Suggestion

Provide small complete code samples in [docs](https://getuikit.com/docs/).  e.g., the `uk-tab` shows only the `tab`.  It would be helpful to show code sample of `uk-tab` and the related `content`.   

```
<ul uk-tab>
    <li class="uk-active"><a href="#">Left</a></li>
    <li><a href="#">Item</a></li>
    <li><a href="#">Item</a></li>
    <li class="uk-disabled"><a href="#">Disabled</a></li>
</ul>
```

[foundation tabs](http://foundation.zurb.com/sites/docs/tabs.html) is an example of showing both the `tab` and `content`.  By the way, 2 or 3 (tab --> content) sets is sufficient, 6 is too many.  

### UIkit version

3.0.0-beta.27

**Comment by vuchl** [comment]
I don't want to argue in here but i think the docs and the tests are pretty documenting on how to use the components. The provide a small overview on how the markup should be structured and what css classes modify the component and last but not least the javascript section implies all events.

**Comment by zzseba78** [comment]
I think the docs are quite clear with lot of examples. In the case you mention, Tabs, the tabs are just the navigation, it works in conjuction with the Switcher.
In the Switcher section you will see a example with Switcher and Tab.
https://getuikit.com/docs/switcher#switcher-and-tab

**Comment by saun4app** [comment]
@vuchl,  thanks for your prompt replay. I agree with you that, `uikit docs` has very high quality.  `uikit docs`  also show signs of [Curse of knowledge](https://en.wikipedia.org/wiki/Curse_of_knowledge); result in requiring more effort from `new users` than you may expect to jump start using `uikit`.

Paying more attention to the "last-mile" or "last-one-percent" effort would make a significant different for a new user.

@zzseba78,

You are right that https://getuikit.com/docs/switcher#switcher-and-tab is a great a *complete enough* example as I suggested.  This example does not stand out in the https://getuikit.com/docs/tab page.  A new user, who see the `uk-tab` example for the first time, is not familiar with `uikit` enough to look for `uk-tab` examples in https://getuikit.com/docs/switche page.

Before you pointed it out.  I did not know to look for  [switcher-and-tab](https://getuikit.com/docs/switcher#switcher-and-tab) while I was looking at [tab](https://getuikit.com/docs/tab).

Placing [switcher-and-tab](https://getuikit.com/docs/switcher#switcher-and-tab) in https://getuikit.com/docs/tab page is an example of the  "last-one-percent"  effort that has significant impact on a new user's effort in learning `uikit`.

I truly want to see `uikit` to become more popular, therefore more sustainable.  The last thing I want to pointlessly arguing with you.  Please don't feel the need to reply this message.

**Comment by zzseba78** [comment]
@saun4app I think it´s quite clear, first paragraph on Tab Component docs:
>The tab component consists of clickable tabs, that are aligned side by side in a list. Its JavaScript functionality extends the Switcher component and is necessary to dynamically transition through different contents using the tabbed navigation.

The weak part of the docs are related to js, it´s a frecuently requested point to elaborate, but the rest it´s concise and clear.

Of course if you have doubts and questions, go here:
https://gitter.im/uikit/uikit

**Comment by saun4app** [comment]
@zzseba78 ,

[Don’t Make Me Think](https://www.sensible.com/dmmt.html) or read;

**Comment by zzseba78** [comment]
@saun4app Well, this phrase maybe it´s great for usability, but if you want to learn something you, you will have to read and think... a lot :)
Good luck.

**Comment by aarongerig** [comment]
@saun4app Do you know the term "RTFM" a.k.a. "Read The Fucking Manual"? It exists for a reason. 😉 Don't expect to just watch a video or look at examples and learn a whole (frontend) framework. Some people always state "don't make me read something", but it doesn't get much clearer explainig something than reading and looking at examples. UIkit docs do exactly that.

**Comment by saun4app** [comment]
Well, I sincerely believe that [uk-tab](https://getuikit.com/docs/tab) is better than [bootstrap tabs](http://getbootstrap.com/javascript/#tabs) and [fundation tabs](http://foundation.zurb.com/sites/docs/tabs.html).  I failed to understand why you guys resist putting a good code sample in [switcher-and-tab](https://getuikit.com/docs/switcher#switcher-and-tab) page, that you already have, on [tab](https://getuikit.com/docs/tab) page, in order to  get more new readers understand `ui-tab` more easily; even though you disagree on how a user *should* read your docs.

Ask yourself, but don't answer, this question.  When you read a doc (of a framework) do you prefer look at code samples or paragraph of text?

I personally prefer to scan code samples on a specific component (e.g, tab) when I look at a new framework.  If I don't see code example(s) that is better than the frameworks that I knew, I would skip the page (component) or the framework.  `tab` is usually my first test case.

Basically, `tab` is a familiar term for most readers.  The good `tab` code sample is not on [tap](https://getuikit.com/docs/tab) page.  `switcher` is not a common enough concept (term).  Do you really expect, readers of `uk-tab` to know to look for `uk-tab` code sample on [switcher-and-tab](https://getuikit.com/docs/switcher#switcher-and-tab) page?

@aarongerig, 
- I failed to see the reference to " just watch a video " or "learn a whole (frontend) framework" from my comment.
- I am pointing specifically to the [uk-tab](https://getuikit.com/docs/tab) not  UIkit docs  in general.  I did say "More than 90% of the code samples are great. "
-  Where I came from,  "RTFM" a.k.a. "Read The Fucking Man Page".

**Comment by aarongerig** [comment]
@saun4app Sorry if you thought my comment was harsh, but a lot of people complain about something that is already there and are too lazy to have a real good look at it. I was more generalizing that I should of. Me myself changed frameworks about half a year ago (came from Bootstrap) and I never had a problem reading the docs until I understood all the specialities of UIkit.

Concerning the `uk-tab` you were talking about:
I think UIkit keeps `tab` and `switcher` apart because of [SoC](https://en.wikipedia.org/wiki/Separation_of_concerns), since the framework's components rely on each other. `uk-tabs` are for styling and `uk-switcher` for toggling the tab.

If you have any more questions about UIkit, either ask on [Gitter](https://gitter.im/uikit/uikit) as @zzseba78 said or hit me up with a PM. Have fun using UIkit!

**Comment by Riddler84** [comment]
Pretty much what @aarongerig said. The tab component is only there for styling purposes. The functionality comes with the switcher component, which can switch not only tabs, but also buttons, navs and subnavs. 

So it's logical to split these things, because you can also use a tab menu without the switcher component, by using the tab menu items as normal links.

And the first paragraph on the tab page has a big hint on the switcher component and explains how they work together. ( mentioned before by @zzseba78 )
I think this makes it clear enough. A bit reading could be expected, couldn't it? ;)

**Comment by dimitrilongo** [comment]
> Before you pointed it out. I did not know to look for switcher-and-tab while I was looking at tab.

i don't understand in https://getuikit.com/docs/tab the first paragraph is pretty clear and even with the link for https://getuikit.com/docs/switcher

>Usage
>
>The tab component consists of clickable tabs, that are aligned side by side in a list. Its JavaScript >functionality extends the Switcher component and is necessary to dynamically transition through >different contents using the tabbed navigation.

You can event click on Test  https://getuikit.com/assets/uikit/tests/switcher.html, every page has a test link.
Then  just explore code source

**Comment by zzseba78** [comment]
@saun4app Take your time to learn the framework and you will see that working with UIkit 3 it´s great. I reccomend to use it with less/sass so you can take the best path from the beginning. Good luck.

**Comment by florianletsch** [comment]
It's great to read this discussion and see that many people seem quit content with how we've structured the docs. Cheers for that, we've put quite a lot of effort into them. :)

### Yes!

I agree with all of your points: Yes, many "new" users might think of "tabs" and want to use them with content immediately. More extensive examples are always great to get people started. That is why in UIkit 3 docs we have These "Example" boxes where you can toggle the complete markup of an example and even view it on Codepen (in UIkit 2, the examples were simplified markup to only show the general structure of a component - they were not copy-paste ready). 

### And here's why:
At the same time, UIkit is very modular and our first priority when writing the docs was to show in a technical way what every component is capable of. Wherever useful, we have added pointers to related components (for example from Tab to Switcher) and the Test of a component usually contains many examples of a component in connection with other components.

### So, I suggest:

Instead of expanding the current examples, I would rather like to include more "Getting Started" information for new users. This would include complete layout examples (like [we had in UIkit 2](https://getuikit.com/v2/docs/documentation_layouts.html)) and some more tutorial style docs (like the [screencasts](https://getuikit.com/v2/docs/tutorials.html) I produced back in the day). Then again, time is limited, so we'll see how soon this can be tackled.

We will take your feedback into account for further improvement. Also, we are currently re-working the YOOtheme Pro documentation, and we will take all of these ideas into account as well. Working on the Pagekit docs, UIkit 2 and 3 docs and [YOOtheme Pro](http://yootheme.com/pro/) docs has slowly improved the way we write, structure and visualize documentation, I'd dare to say. ;) It's an iterative process, so your feedback is always highly appreciated.

**Comment by saun4app** [comment]
@florianletsch, I appreciate your thoughtful comment.   I spent significant amount of time commenting here (and https://gitter.im/uikit/uikit) is because I think `uikit` (the freamework and document) is better than most CSS frameworks out there.  With *very little effort*, adding existed code samples, will enable new site visitors to get to know `uikit` with less effort,

**Comment by saun4app** [comment]
@florianletsch, here are some thoughts (comments) related to your "So, I suggest" section with "time is limited" in mind.

- Currently, `uikit` docs offers sufficient (or very close to sufficient) *number* of code samples.  Placing code samples on the same page as the UI component (without increasing number of samples) would be helpful. 

- In programming setting, many (some, most) of CSS framework target user are likely to be more proficient in `code` languages (CSS, HTML, JS, etc.) than in English.  Therefore, 

1. showing code sample of a UI component (with resulting screen short or preview, like uikit docs are doing) is likely to be more effective (and require less effort from the user) than describing the UI components in English or screen casts.  I suspect that a typical CSS framework target user can understand   [switcher-and-tab](https://getuikit.com/docs/switcher#switcher-and-tab) example with previews in around one minute; and would take longer with English description or a screen cast.

1. the focus of code samples is less about "Getting Started" vs. "intermediate, advance" code samples; it is more about demonstrating multiple (nearly all) usages of a UI component.  Using tab as can example, the multiple usages may include horizontal, vertical, responsive, *font setting*, *content display types* (li, dev) etc. 

1. most CSS framework target users know what UI features to look for.  They just need to see (and re-reference) the syntax for the particular (new) framework for implementing the mostly known (familiar) UI features. [uikit docs](https://getuikit.com/docs) is one of the best examples that I know of that offers one document that serves as introductory and reference docs.  Given the target user and the nature of CSS framework, I think the current structure of uikit docs is great.  [Foundation](http://foundation.zurb.com/) is a counter example.  Foundation uses a lot of screen casts.  Viewing the screen casts is time consuming.  Reference back to the code samples is hopeless. 

- [tab](https://getuikit.com/docs/tab) is a UI component.  [switcher](https://getuikit.com/docs/switcher) is not really a UI component; switcher appears to be more of a very useful utility feature, (function, capability, helper) of number of UI components.   switcher appears to be documented as if it was a UI component;  A user new *to uikit* (not new to coding) likely knows to look for a UI components (e.g., tab), and know to look for typical (and special) tab features; BUT not know enough to look for switcher.  Therefore, all tab code samples should appear on tab page.  

In short, by mostly *repurposing* existing code samples can go a long way to enable new site visitors to get to know `uikit` with less effort; without putting significant time and effort in crating new samples and screen casts.

**Comment by georgykarpenko** [comment]
I'm totally agree with @saun4app, every component starts with small simple example, so why not to write a complete working example? Tabs are used to switch between content, so show this ...

I have found this issue because was also looking for "How to use UIkit Tabs"

**Comment by irekjozwiak** [comment]
I completely agree with @saun4app . A working example for tabs would be very useful.

Remember that the relationship between tabs and switcher is something that is very clear for the authors of UIkit and more advanced users. It is not necessarily clear for other users (like myself).

I found this discussion because I'm still trying to make tabs work for me. Without an example, the way to go for me was to find actual tabs on a UIkit page and "inspect" them in the browser to see how it's actually done. I still haven't got it working but hopefully I'll get there. A working example really wouldn't hurt. Certainly it would save me time.
